examples/decode.rb

```diff
@@ -0,0 +1,31 @@
+module Authorization
+  #validation using JWT gem
+  def validate_jwt
+    return if jwt_valid?
+    render json: error
+  end
+
+  def jwt_valid?
+    @decoded_jwt ||=
+      begin
+        JWT.decode(
+        jwt, 
+        public_key, 
+        true, 
+        algorithm: "RS256", 
+        iss: "expected_issuer", 
+        verify_iss: true)
+      rescue JWT::DecodeError => e
+        log(e)
+      end
+    @decoded_jwt.present?
+  end
+
+  def jwt
+    request.authorization.to_s[/(?<=\ABearer) +.*\z/].try!(:strip)
```

Comment:
Here's another Rails abstraction we can make explicit in this example code:  https://github.com/rails/rails/blob/1a240cf83e03527567dea68f3c6bea5301d90113/actionpack/lib/action_dispatch/http/request.rb#L404